src/display/canvas.js

```diff
@@ -956,6 +956,15 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
             break;
         }
       }
+
+      if (typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) {
```

Comment:
I'm also a bit nervous about this. Would checking if `globalAlpha` is set to a non-default value before overwriting it address this concern? I do think that it would be best to consolidate this workaround in `compatiblity.js` and not have any changes in `canvas.js` to avoid any risks.

Reply:
@Snuffleupagus, The reason why I override the alpha value in `canvas.js` as opposed to the compatibility file is to avoid my value being overriden by a later `ca` instruction. I noticed that the order of instruction type is not guaranteed. 

I agree with @timvandermeij that I should include a check for a current global alpha value before setting hard coding it myself. An example of this problem would be `BM = 'overlay'` and `ca = 0.25`. My current implementation would overwrite this to 0.5, but since all content would be visible it should be left as is. 

I'll attempt to access this from the compatibility file but it may only be accessible after we have processed all of our instructions in `canvas.js` (where I am currently overriding the alpha value now). I'll include a new commit with the suggested style fixes + my proposed safety change.